src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4876,6 +4876,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case LookupInaccessible(sym, msg) => issue(AccessError(tree, sym, context, msg))
           case LookupNotFound               =>
             inEmptyPackage orElse lookupInRoot(name) match {
+              // ignore wildcard bodies inserted by the Java parser
+              case NoSymbol if context.unit.isJava && name == termNames.WILDCARD => tree
```

Comment:
I get the compilation errors like these 
```scala
JavaClazz.java:20: not found: value _
    public void foo() {}
                ^
```
They appear at all places where the java parser inserts a `_` as placeholder body, for instance in method and field bodies and parameters.